.github/workflows/main.yml

```diff
@@ -93,24 +103,34 @@ jobs:
     needs: [test]
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: "3.12"
+          version: "0.4.12"
+          enable-cache: true
+
+      - name: Set up Python 3.12
+        run: uv python install 3.12
+
+      - name: Create venv
+        run: uv venv --python 3.12
```

Comment:
Why do we create a venv here but not elsewhere? Are we using it? Can it be activated to avoid needing `uv run`?

Reply:
Because everywhere else we run `uv sync`.